### PR TITLE
Homework #1

### PR DESCRIPTION
**Pull request by guardspb**
I did homework and everything works good.

### CODE REVIEW

File: homework1/task05.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    """Returns the maximal sum of subarray with length less or equal to "k"."""
+
+    if not nums or not k:
+        return 0
+
+    n = len(nums)
+
+    sub_sum, res_sum = 0, nums[0]
+
+    for j in range(min(k, n)):
```
Comment (nikitosboroda) [inline homework1/task05.py:15]: if k > n - should be an error

File: homework1/task05.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
```
Comment (nikitosboroda) [inline homework1/task05.py:4]: what will be if k<0?

File: homework1/task02.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from math import sqrt
+from typing import Sequence
+
+
+def check_fibonacci(data: Sequence[int]) -> bool:
```
Comment (nikitosboroda) [inline homework1/task02.py:5]: 1) there could be much easier algorithm
2) let it be the fibb seq only when length of seq >=3

File: tests/homework1/test_task02.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        [0, 1],
+        [1, 1],
+        [5, 8],
+        [8, 13, 21],
+        [13, 21, 34, 55, 89],
+        [0, 1, 1, 2, 3, 5, 8, 13],
+    ],
+)
+def test_positive_cases_lists(data):
+    assert check_fibonacci(data) is True
```
Comment (nikitosboroda) [inline tests/homework1/test_task02.py:22]: u dont need ```is True``` there

File: tests/homework1/test_task02.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        (34,),
+        (0, 1),
+        (1, 1),
+        (5, 8),
+        (8, 13, 21),
+        (13, 21, 34, 55, 89),
+        (0, 1, 1, 2, 3, 5, 8, 13),
+    ],
+)
+def test_positive_cases_tuples(data):
```
Comment (nikitosboroda) [inline tests/homework1/test_task02.py:40]: what's the difference with previous tests? list\tuple - doesnt matter

File: tests/homework1/test_task02.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        [0, 0],
+        [3, 3],
+        [5, 7],
+        [5, 8, 23],
+        [5, 8, 23, 21, 55],
+        [0, 1, 1, 0, 3, 8, 21, 89],
+    ],
+)
+def test_negative_cases_lists(data):
+    assert check_fibonacci(data) is False
```
Comment (nikitosboroda) [inline tests/homework1/test_task02.py:79]: the same about ```is```

**Review by nikitosboroda — CHANGES_REQUESTED** [review]

**Comment by guardspb** [comment]
Good evening, dear mentor! I have corrected all tasks according to your remarks.

**Comment by guardspb** [comment]
Good evening, dear mentor! I sent you the pull request.

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -17,6 +17,6 @@ jobs:
         run: |
           pip install flake8 isort pytest
           ls -la
-          flake8 .
+          flake8 --max-line-length=120 .
```
Comment (nikitosboroda) [inline .github/workflows/main.yml:20]: set max 88.. if your row bigger than 88 symbls, so split it

File: homework1/task02.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
```
Comment (nikitosboroda) [inline homework1/task02.py:5]: what will be if ```data = [1, 10]``` or ```data = [1]```?

File: homework1/task05.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+    """Returns the maximal sum of subarray with length less or equal to "k"."""
+
+    if not nums or not k:
+        return 0
```
Comment (nikitosboroda) [inline homework1/task05.py:9]: ur func returns integer.. so user couldnt understand that he got ```0``` cause some problems with ```nums``` or ```k```.. he could think 0 - is the max sum

File: tests/homework1/test_task05.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import pytest
+
+from homework1.task05 import find_maximal_subarray_sum
+
+
```
Comment (nikitosboroda) [inline tests/homework1/test_task05.py:5]: what about negative cases?

**Review by nikitosboroda — CHANGES_REQUESTED** [review]

**Comment by guardspb** [comment]
Good evening, dear mentor! I have corrected all tasks according to your remarks.

**Review by nikitosboroda — APPROVED** [review]

**Pull request merged**